point-of-sale-app/inventory/src/main/java/com/google/abmedge/inventory/dto/PurchaseItem.java

```diff
@@ -0,0 +1,38 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.abmedge.inventory.dto;
+
+import java.util.UUID;
+
+public class PurchaseItem {
+  private UUID itemId;
+  private long itemCount;
+
+  public UUID getItemId() {
```

Comment:
Why not have a constructor and avoid setter methods. PurchaseItem can change UUID and itemCount? Also the mutability of PurchaseItem is a question which can direct you constructor vs setter method way.

Reply:
There is no place an object of this class is explicitly created by our code. The only place it is used in is the function argument to the [`update` API of the inventory controller](https://github.com/GoogleCloudPlatform/anthos-edge-usecases/blob/add-inventory-apis/point-of-sale-app/inventory/src/main/java/com/google/abmedge/inventory/InventoryController.java#L134). So this class is used by Springboot to deserialize a json request body into a java object when an API request arrives at /update. The serializer requires the default constructor and getter/setters.